hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/KeyGenUtils.java

```diff
@@ -66,29 +66,25 @@ public class KeyGenUtils {
   public static KeyGeneratorType inferKeyGeneratorType(
       Option<String> recordsKeyFields, String partitionFields) {
     boolean autoGenerateRecordKeys = !recordsKeyFields.isPresent();
-    if (autoGenerateRecordKeys) {
-      return inferKeyGeneratorTypeForAutoKeyGen(partitionFields);
-    } else {
-      if (!StringUtils.isNullOrEmpty(partitionFields)) {
-        int numPartFields = partitionFields.split(",").length;
-        int numRecordKeyFields = recordsKeyFields.get().split(",").length;
-        if (numPartFields == 1 && numRecordKeyFields == 1) {
-          return KeyGeneratorType.SIMPLE;
-        }
-        return KeyGeneratorType.COMPLEX;
-      }
-      return KeyGeneratorType.NON_PARTITION;
+    KeyGeneratorType partitionKeyGeneratorType = inferKeyGeneratorTypeFromPartitionFields(partitionFields);
+    if (autoGenerateRecordKeys || partitionKeyGeneratorType != KeyGeneratorType.SIMPLE) {
```

Comment:
Why is `|| partitionKeyGeneratorType != KeyGeneratorType.SIMPLE` added?

Reply:
Should `KeyGeneratorType.COMPLEX` be included too based on the logic below.

Reply:
KeyGeneratorType is SIMPLE only if both partition fields and record key fields have one field configured. For that reason we need additional check to know how many record key fields are present apart from partition fields. 
For other partition types, we can infer based on partition type alone. This is aligned with older logic.

Reply:
Make sense.  In that case should all logic of checking record key and partition fields be put in the same method `inferKeyGeneratorTypeFromPartitionFields`?  The code in the current shape is confusing.  Still keep the structure like:
```
if (autoGenerateRecordKeys) {
      return inferKeyGeneratorTypeForAutoKeyGen(partitionFields);
}
...
```

Reply:
Addressed. I have tried to simplify the logic further. Please take a look.